src/physlr-map.cc

```diff
@@ -0,0 +1,530 @@
+
+		for (uint64_t queryPos = 0; queryPos < minimizers.size(); ++queryPos) {
+			auto& minimizer = minimizers[queryPos];
+			if (minimizerToPos.find(minimizer) != minimizerToPos.end()) {
+				auto& vectorTargetIdPos = minimizerToPos.at(minimizer);
+				for (const auto& targetIdPos : vectorTargetIdPos) {
+					if (targetIdPosToQuerypos.find(targetIdPos) == targetIdPosToQuerypos.end()) {
+						targetIdPosToQuerypos[targetIdPos] = std::vector<uint64_t>();
+						targetIdPosToQuerypos[targetIdPos].emplace_back(queryPos);
+					} else {
+						targetIdPosToQuerypos[targetIdPos].emplace_back(queryPos);
+					}
+				}
+			}
+		}
+
+		for (const auto& targetIdPosToQueryposKeyVal : targetIdPosToQuerypos) {
+
+			auto& targetIdPos = targetIdPosToQueryposKeyVal.first;
+			auto& queryPos = targetIdPosToQueryposKeyVal.second;
+
+			targetIdPosToQuerypos[targetIdPos] = { lowerMedian(queryPos) };
+		}
+
+		tsl::robin_map<pair, uint64_t, boost::hash<pair>> targetIdPosToCount;
+		for (const auto& minimizer : minimizers) {
+			if (minimizerToPos.find(minimizer) != minimizerToPos.end()) {
+				auto& vectorTargetIdPos = minimizerToPos.at(minimizer);
+				for (const auto& targetIdPos : vectorTargetIdPos) {
+					if (targetIdPosToCount.find(targetIdPos) == targetIdPosToCount.end()) {
+						targetIdPosToCount[targetIdPos] = 1;
+					} else {
+						++targetIdPosToCount[targetIdPos];
+					}
+				}
+			}
+		}
+
+		bool mapped = false;
+
+		for (const auto& targetIdPosToCountKeyVal : targetIdPosToCount) {
+
+			auto& targetIdPos = targetIdPosToCountKeyVal.first;
+			auto& score = targetIdPosToCountKeyVal.second;
+
+			if (score >= opt::scoreThreshold) {
+
+				mapped = true;
+				auto& targetId = std::get<0>(targetIdPos);
+				auto& targetPos = std::get<1>(targetIdPos);
+				uint64_t max = std::numeric_limits<std::uint64_t>::max();
+				std::string orientation;
+				uint64_t prev;
+				uint64_t next;
+				uint64_t curr = targetIdPosToQuerypos[targetIdPos][0];
+
+				if (opt::mapPositions == 1) {
+
+					auto prevPair = std::make_pair(targetId, targetPos - 1);
+					auto nextPair = std::make_pair(targetId, targetPos + 1);
+
+					if (targetIdPosToQuerypos.find(prevPair) == targetIdPosToQuerypos.end()) {
+						prev = max;
+					} else {
+						prev = targetIdPosToQuerypos[prevPair][0];
+					}
+
+					if (targetIdPosToQuerypos.find(nextPair) == targetIdPosToQuerypos.end()) {
+						next = max;
+					} else {
+						next = targetIdPosToQuerypos[nextPair][0];
+					}
+				} else {
+
+					std::vector<uint64_t> prevVec;
+
+					for (unsigned i = 1; i <= opt::mapPositions; ++i) {
+						auto prevPair = std::make_pair(targetId, targetPos - i);
+						if (targetIdPosToQuerypos.find(prevPair) != targetIdPosToQuerypos.end()) {
+							prevVec.emplace_back(targetIdPosToQuerypos[prevPair][0]);
+						}
+					}
+
+					if (prevVec.empty()) {
+						prevVec.emplace_back(max);
+					}
+					prev = lowerMedian(prevVec);
```

Comment:
Don't you do a check if `prevVec` is empty in `lowerMedian`? Do you still need to add `max` to the `prevVec` then?

Reply:
This is for determineOrientation. I can either do what I did up there or do this 
`if (prevVec.empty()) {
determineOrientation(max, curr, next);
}` and 2 more for the other possibilities.

There is more code duplication for the latter

Reply:
Could you just return `max` from `lowerMedian`/`upperMedian` if prevVec is empty?

Reply:
Hmm yes, but median function giving me a max value when the input array is empty doesn't feel quite right.

Reply:
I'm only checking the vector once.